sampler.go

```diff
@@ -211,21 +210,25 @@ type GuaranteedThroughputProbabilisticSampler struct {
 func NewGuaranteedThroughputProbabilisticSampler(
 	lowerBound, samplingRate float64,
 ) (*GuaranteedThroughputProbabilisticSampler, error) {
-	tags := []Tag{
-		{key: SamplerTypeTagKey, value: SamplerTypeLowerBound},
-		{key: SamplerParamTagKey, value: samplingRate},
+	s := &GuaranteedThroughputProbabilisticSampler{
+		lowerBoundSampler: NewRateLimitingSampler(lowerBound),
+		lowerBound:        lowerBound,
 	}
-	probabilisticSampler, err := NewProbabilisticSampler(samplingRate)
-	if err != nil {
-		return nil, err
+	s.setSamplingRate(samplingRate)
+	return s, nil
+}
+
+func (s *GuaranteedThroughputProbabilisticSampler) setSamplingRate(samplingRate float64) {
+	if s.probabilisticSampler == nil || s.samplingRate != samplingRate {
+		probabilisticSampler, _ := NewProbabilisticSampler(samplingRate)
```

Comment:
will `go vet` not complain about swallowing the error? That's the reason I was suggesting a new "safe" constructor.